Improving unit test stability on Travis CI

PhilMFischer opened the pull request:
Initially the plan was to:

- Reduced logging size, since logs are restricted to 4MB
- Running unit tests for eac individual unit test fragment rather than all togeter in overall tesuite. Still the overall testuite is needed for downstream projects and fo executing tests in eclipse.
- Fixed issues with tests, and test confusion. every concept or plugin should have a test fragment. AllTests is suite for tests. In particular the surefire settings are realy sensitive to these settings. 
- Fixed several dependency issues  which became visible with the individual test execution.

Closes #250

Further along the way:
- We had to deal with checksum issues (Closes #253)
- Worked on dependecies on AProjectTestCase. Replaced all usages with AConceptProjectTestCase. Moving as suggested by #242 is not possible. (Closes #242)
- Fixed a bug on VirSatEditingDomain regarding the notification thread. In some test cases the thread was stopped making the test cases get stuck. Could be a reason to general test case instability we have seen in the past.
- Fixed issues with mechanical.test execution. Due to the individual teste executions  Super User Rightsw were not set, and therefore the inheritance copier failed to work as expected. Changed AProjectTestCase to now have SuperUser rights always being set. It seems to be the general case for tests.

Commit:
Disabled Logging for Teste Execution on Maven - (Task #250)

- Disabled in Master Pom
- Disabled in SwtBOT tests Pom

---
Task #250: Disable Logging on Travis CI

Commit:
Renaming calculation test plugin - (Task #250)

- Calculation test was called ...tests and should eb called ...test

---
Task #250: Disable Logging on Travis CI

Commit:
Minor fix to feature xml due to renaming - (Task #250)

- Test feature needs to reference caluclation.test now

---
Task #250: Disable Logging on Travis CI

Commit:
Now executing tycho tests directly in test plugin

Commit:
Added example test case in team - (Task #250)

- Added example test case to not make travisCI fail
- Fixed remains of calulcation.tests which should eb called test

---
Task #250: Disable Logging on Travis CI

Commit:
Merge branch 'feature/task_250-Disable_Logging_on_Tr' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_250-Disable_Logging_on_Tr

Commit:
Minor fix to feature xml due to renaming - (Task #250)

- POM was not renamed yet

---
Task #250: Disable Logging on Travis CI

Commit:
Fixing dependency to ANTLR - (Task #250)

- Xtext requires 3.2.x version 4 is not yet supported and is not allowed
to be part of the target platform. Otherwise it may overrule the older
version and execution of testcases are going to fail

---
Task #250: Disable Logging on Travis CI

Commit:
Adding timeouts and retries to SWTBOT- (Task #250)

- Added a timeout of 90 seconds to each individual test case of SWTBOT
- Running SWTBOT from surefire with 3 retries until it is regarded as a
complete fail.

---
Task #250: Disable Logging on Travis CI

Commit:
Moved DVLM latest Concept Validator test - (Task #250)

- The Validator needs the tests model extension to be laoded. This
cannot be guaranteed from the build.test plugin.

---
Task #250: Disable Logging on Travis CI

Commit:
Fixed issue with PropertyInstanceHelper TestCase - (Task #250)

- Added default behavior to isCalculated Method
- Externalized loading of extension point
- Added test case with mocked extension point to check correct behavior

---
Task #250: Disable Logging on Travis CI

Commit:
Switching surefire to a fail early approach - (Task #250)

- Surefire will no stop and fail the build once a test fails.

---
Task #250: Disable Logging on Travis CI

Commit:
Adjusted Warning detection - (Task #253)

 - Excluding Checksum validation erors
---
Task #253: Don't react to checksum errors on build

Commit:
Merge branch 'feature/task_253-Don't_react_to_checks' into feature/task_250-Disable_Logging_on_Tr

Commit:
Further fixing to dvlm latest version validator - (Task #250)

- Refactored further incorrect dependencies

---
Task #250: Disable Logging on Travis CI

Commit:
Remove warnings - (Task #250)

Put your Message here.

---
Task #250: Disable Logging on Travis CI

Commit:
fixed team example test case - (Task #250)

 - made test method public

---
Task #250: Disable Logging on Travis CI

Commit:
Set SWTBOT test also to fail early - (Task #250)

- Now where SWT Bot is running three times to push for a positive
result, it should fail early to make developers easily spot what is
going wrong.
- Rename AllTest for swtbot to AllTests, thus it will not get included
to the surefire test lasses. It is not in the standard include pattern
of surefire
- Adjusted Launch config for SWTBOT tests running from eclipse.

---
Task #250: Disable Logging on Travis CI

Commit:
Merging with Development - (Task #250)

- Since test cases are now collected and individually executed by
surefire, the Test name watcher is not needed anymore.

---
Task #250: Disable Logging on Travis CI

Commit:
Merge branch 'development' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_253-Don't_react_to_checks

SaMuellerDLR commented on de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java:
```diff
@@ -32,9 +32,13 @@
 import org.junit.rules.TestName;
+<<<<<<< HEAD
+import org.junit.rules.Timeout;
+=======
```
Weird merge remains.

PhilMFischer replied:
Fixed. Did not pay attention when merging.

SaMuellerDLR commented on de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java:
```diff
@@ -80,6 +84,11 @@ protected void finished(Description description) {
+	protected static final int MAX_TEST_CASE_TIMEOUT_SECONDS = 90;
+	
+	@Rule
+    public Timeout globalTimeout = Timeout.seconds(MAX_TEST_CASE_TIMEOUT_SECONDS);
```
Weird whitespacing.

PhilMFischer replied:
Indeting was mixed by ws and tab. Fixed it.

Commit:
Fixing incorrect merge - (Task #250)

- ASwtbBottestCase got corrupted while merging.

---
Task #250: Disable Logging on Travis CI

Commit:
Fixed indeting - (Task #250)

- Mix of whitespace vs tabs.

---
Task #250: Disable Logging on Travis CI

SaMuellerDLR reviewed (approved):
Approved!

Commit:
Merge branch 'feature/task_253-Don't_react_to_checks' into feature/task_250-Disable_Logging_on_Tr

Commit:
Trying to tune build time - (Task #250)

- Replaced javadoc with doc profile
- Doc profile now responsible for javadoc and Asciidoc generation
- doc profile now gets executed in assembly only

---
Task #250: Disable Logging on Travis CI

Commit:
Speeding up travis builds - (Task #250)

- Now only building p2 and features when needed for deploy. Checkstyle
spotbugs and surefire now act on plugins only. Therefroe only plugins
are built for these profiles.

---
Task #250: Disable Logging on Travis CI

Commit:
Trying to fix warnings about local dependencies - (Task #250)

- Added maven goal clean

---
Task #250: Disable Logging on Travis CI

Commit:
SWTBot does not need dependency to javadoc - (Task #250)



---
Task #250: Disable Logging on Travis CI

Commit:
Added additional checks to CatiaImportertest - (Task #250)

- Check if commands get properly executed.

---
Task #250: Disable Logging on Travis CI

Commit:
Enabling eclispe log for mechaincal test - (Task #250)

- Trying to get mor information about fail.

---
Task #250: Disable Logging on Travis CI

Commit:
Speed up travis to find issue - (Task #250)

- Changed travisCI to just run the surefire tests

---
Task #250: Disable Logging on Travis CI

Commit:
Disabled other test cases - (Task #250)

- Now only builoding the mechanical.test. All others are deactivated.

---
Task #250: Disable Logging on Travis CI

Commit:
Condensed TravisCI to surefire development only - (Task #250)

- All other builds are rmeoved fro the moment

---
Task #250: Disable Logging on Travis CI

Commit:
Trying to see if visualization tests is executing - (Task #250)



---
Task #250: Disable Logging on Travis CI

Commit:
Reactivating mechanical test - (Task #250)

---
Task #250: Disable Logging on Travis CI

SaMuellerDLR committed:
Simplified funcelectircal excel test cases - (Task #250)


---
Task #250: Disable Logging on Travis CI

SaMuellerDLR committed:
Merge branch 'feature/task_250-Disable_Logging_on_Tr' of https://github.com/virtualsatellite/VirtualSatellite4-Core into feature/task_250-Disable_Logging_on_Tr

Commit:
Moving is not enough- (Task #242)

- Did a Copy and paste of the TestClass. Adjusted dependencies to
AProjectTestCase

---
Task #242: Moving AProjectTestCase to concept.unittest.utils

Commit:
Merge branch 'feature/task_250-Disable_Logging_on_Tr' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_250-Disable_Logging_on_Tr

Commit:
Deactivating further tests - (Task #242)

- Forgot to deactivate model.test

---
Task #242: Moving AProjectTestCase to concept.unittest.utils

Commit:
Further checks to test data - (Task #250)

- Make sure test data is not corrupted

---
Task #250: Disable Logging on Travis CI

Commit:
Further checks to test data - (Task #250)

- Make sure test data is not corrupted

---
Task #250: Disable Logging on Travis CI

Commit:
file with surefire call - (Task #250)

- Copied the call for surefire on Travis to a file. trying to reproduce
problem.

---
Task #250: Disable Logging on Travis CI

SaMuellerDLR committed:
Trying out super user - (Task #250)


---
Task #250: Disable Logging on Travis CI

SaMuellerDLR committed:
Outputting user name - (Task #250)


---
Task #250: Disable Logging on Travis CI

Commit:
Debugging Test Failures - (Task #250)


---
Task #250: Disable Logging on Travis CI

Commit:
Merge branch 'feature/task_250-Disable_Logging_on_Tr' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_250-Disable_Logging_on_Tr

Commit:
Debugging Test Failures - (Task #250)


---
Task #250: Disable Logging on Travis CI

Commit:
Debugging Test Failures - (Task #250)


---
Task #250: Disable Logging on Travis CI

SaMuellerDLR committed:
Properly toggling super user rights - (Task #250)


---
Task #250: Disable Logging on Travis CI

SaMuellerDLR committed:
merge - (Task #250)

---
Task #250: Disable Logging on Travis CI

Commit:
Reneabled all unit tests

Commit:
Merge branch 'feature/task_250-Disable_Logging_on_Tr' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_250-Disable_Logging_on_Tr

SaMuellerDLR committed:
Made ExecutionCheckCommand a general test helper class - (Task #250)


---
Task #250: Disable Logging on Travis CI

SaMuellerDLR committed:
Merge branch 'feature/task_250-Disable_Logging_on_Tr' of https://github.com/virtualsatellite/VirtualSatellite4-Core into feature/task_250-Disable_Logging_on_Tr

SaMuellerDLR committed:
Removing debug data - (Task #250)


---
Task #250: Disable Logging on Travis CI

SaMuellerDLR committed:
Cleaned up test pom - (Task #250)


---
Task #250: Disable Logging on Travis CI

Commit:
Debugging Test Failures - (Task #250)

- Applied several fixes to Test for Virsat transactional Editing Domain
- Removed bug regarding accumulated Resource Event Thread creation and
re-creation
- Refactored AProjectTestcase back to old settings

---
Task #250: Disable Logging on Travis CI

Commit:
Merge branch 'feature/task_250-Disable_Logging_on_Tr' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_250-Disable_Logging_on_Tr

Commit:
Replaced AProjectTestCase - (Task #250)

- A ProjectTestCase should only be used in project.test. Outside the
AConceptProjectTestCase should be used instead

---
Task #250: Disable Logging on Travis CI

Commit:
Fix jacoco report collection - (Task #250)

- Avoiding javadoc plugin for jacoco report generation.

---
Task #250: Disable Logging on Travis CI

Commit:
Bring back all travis stages - (Task #250)

---
Task #250: Disable Logging on Travis CI

Commit:
Updating launcher profile - (Task #250)

- Maven build has a new doc profile which replaces the javadoc profile

---
Task #250: Disable Logging on Travis CI

Commit:
Fix spotbugs warning - (Task #250)

- Make creation of thread threadsafe :-D

---
Task #250: Disable Logging on Travis CI

Commit:
Updating JavaDoc generation - (Task #250)

- Removed test dependencies from javadoc generation

---
Task #250: Disable Logging on Travis CI
